File: examples/typescript/src/command.ts
// ── The signed action (seam 02) ─────────────────────────────────────────────
// A human tap and an agent tool-call become the SAME Command, differing only by
// `by: Actor`. Everything — even a UI focus — is a command on one bus.

import type { Actor, Priority } from "./domain";

export type Command =
  | { readonly kind: "SetPriority"; readonly by: Actor; readonly id: string; readonly ticket: string; readonly level: Priority }
  | { readonly kind: "RequestEscalation"; readonly by: Actor; readonly id: string; readonly ticket: string }
  | { readonly kind: "ConfirmEscalation"; readonly by: Actor; readonly id: string; readonly ticket: string }
  | { readonly kind: "FocusTicket"; readonly by: Actor; readonly id: string; readonly ticket: string };

File: reference-impl/kotlin/src/main/kotlin/adr/Command.kt
// ── The signed action (seam 02) ─────────────────────────────────────────────
// A human tap and an agent tool-call become the SAME Command, differing only by
// `by: Actor`. Everything — even a UI focus — is a command on one bus.

package adr

sealed interface Command {
    val by: Actor
    val id: String
    val ticket: String

    data class SetPriority(
        override val by: Actor,
        override val id: String,
        override val ticket: String,
        val level: Priority,
    ) : Command

    data class RequestEscalation(
        override val by: Actor,
        override val id: String,
        override val ticket: String,
    ) : Command

    data class ConfirmEscalation(
        override val by: Actor,
        override val id: String,
        override val ticket: String,
    ) : Command

    data class FocusTicket(
        override val by: Actor,
        override val id: String,
        override val ticket: String,
    ) : Command
}
